mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -488,10 +491,14 @@ abstract class DecisionTreeTests(sc: SparkContext)
 
   protected var labelType = -1
 
-  def validate(model: RandomForestModel, rdd: RDD[LabeledPoint]): Double = {
+  def validate(model: Either[RandomForestModel, GradientBoostedTreesModel], rdd: RDD[LabeledPoint])
+    : Double = {
     val numExamples = rdd.count()
     val predictions: RDD[(Double, Double)] = rdd.map { example =>
-      (model.predict(example.features), example.label)
+      model match {
```

Comment:
Nope; unlike `Option` there's no well-defined default behavior for the other type in the coproduct. We could `fold` here and pass in two functions (one for `Left` and one for `Right), but IMO that's about as verbose as pattern matching